packages/lib/piiFreeData.ts

```diff
@@ -59,18 +59,12 @@ export function getPiiFreeBooking(booking: {
 }
 
 export function getPiiFreeCredential(credential: Partial<Credential>) {
-  return {
-    id: credential.id,
-    invalid: credential.invalid,
-    appId: credential.appId,
-    userId: credential.userId,
-    type: credential.type,
-    teamId: credential.teamId,
-    /**
-     * Let's just get a boolean value for PII sensitive fields so that we atleast know if it's present or not
-     */
-    key: getBooleanStatus(credential.key),
-  };
+  /**
+   * Let's just get a boolean value for PII sensitive fields so that we atleast know if it's present or not
+   */
+  const booleanKeyStatus = getBooleanStatus(credential?.key);
+
+  return { ...credential, key: booleanKeyStatus };
```

Comment:
Let's just change the credential keys to a boolean and return the credential as we received it.